test/testutil/fakesurvey.go

```diff
@@ -54,62 +64,93 @@ func (m *AskMocker) AsAsker() func(p survey.Prompt, response interface{}, opts .
 
 func NewAskMocker() *AskMocker {
 	return &AskMocker{
-		Question: make(chan survey.Prompt),
-		Answer:   make(chan answerOrError),
+		Question:            make(chan questionWithOptions),
+		Answer:              make(chan answerOrError),
+		LastValidationError: make(chan error),
 	}
 }
 
+func (m *AskMocker) GetLastValidationError() error {
+	return <-m.LastValidationError
+}
+
 func (m *AskMocker) Close() {
 	m.Closed = true
 	close(m.Question)
 	close(m.Answer)
 }
 
-func (m *AskMocker) GetPendingMessageCount() int {
-	return int(m.pendingMsgCount)
-}
-
-func (m *AskMocker) ReceiveQuestion() (survey.Prompt, bool) {
+func (m *AskMocker) receiveQuestion() (survey.Prompt, *survey.AskOptions, bool) {
 	if m.Closed {
-		return nil, false
+		return nil, nil, false
 	}
-	atomic.AddInt32(&m.pendingMsgCount, 1)
 	request := <-m.Question
-	atomic.AddInt32(&m.pendingMsgCount, -1)
-	return request, !m.Closed // reading from closed channels works fine and just returns the default
+	return request.question, request.options, !m.Closed // reading from closed channels works fine and just returns the default
 }
 
-func (m *AskMocker) SendAnswer(answer any, err error) {
+// sendAnswer blindly sends an answer down the answer channel, regardless of whether
+// a question has been asked or not. You should use ExpectQuestion.AnswerWith instead of this
+func (m *AskMocker) sendAnswer(answer any, err error) {
 	if m.Closed {
 		return
 	}
 
-	atomic.AddInt32(&m.pendingMsgCount, 1)
 	m.Answer <- answerOrError{answer: answer, error: err}
-	atomic.AddInt32(&m.pendingMsgCount, -1)
 }
 
-// higher level niceties
-
-func (m *AskMocker) ExpectQuestion(t *testing.T, question survey.Prompt) *QuestionWrapper {
-	q, ok := m.ReceiveQuestion()
+// ReceiveQuestion gets the next question and options from the channel and returns them in a wrapper struct.
+// If the channel is closed, will still return a valid wrapper, but the wrapped contents will be nil
+func (m *AskMocker) ReceiveQuestion() *QuestionWrapper {
+	prompt, askOptions, ok := m.receiveQuestion()
 	if !ok {
 		return &QuestionWrapper{Asker: m}
 	}
-	assert.Equal(t, question, q)
-	return &QuestionWrapper{Question: q, Asker: m}
+	return &QuestionWrapper{Question: prompt, Options: askOptions, Asker: m}
+}
+
+// ExpectQuestion calls ReceiveQuestion and asserts that the received survey prompt matches `question`
+func (m *AskMocker) ExpectQuestion(t *testing.T, question survey.Prompt) *QuestionWrapper {
+	q := m.ReceiveQuestion()
+	assert.Equal(t, question, q.Question)
+	return q
 }
 
 type QuestionWrapper struct {
 	// in case you need it
 	Question survey.Prompt
+	Options  *survey.AskOptions
 	Asker    *AskMocker
 }
 
-func (q *QuestionWrapper) AnswerWith(answer any) {
-	q.Asker.SendAnswer(answer, nil)
+// AnswerWith runs any validators associated with the question. If they all pass, it sends the answer
```

Comment:
This is significant; we now have unit test coverage which covers survey response validators too.
it may not work quite how you may expect, but it's definitely easier and effective this way